From 33bc91cc8e622ef776895cf67f5c8133520961c8 Mon Sep 17 00:00:00 2001
From: Wang Wensheng <wangwensheng4@huawei.com>
Date: Wed, 9 Feb 2022 16:10:56 +0800
Subject: [PATCH 2663/4585] [Huawei] mm/sharepool: Accept device_id in k2u
 flags

Offering: HULK
ascend inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5DS9S

-------------------------------------------------

We use device_id to select the correct dvpp vspace range when SP_DVPP
flag is specified.

Signed-off-by: Wang Wensheng <wangwensheng4@huawei.com>
---
 mm/share_pool.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mm/share_pool.c b/mm/share_pool.c
index e37703dc550c..8f267c649230 100644
--- a/mm/share_pool.c
+++ b/mm/share_pool.c
@@ -3096,10 +3096,11 @@ static int sp_k2u_prepare(unsigned long kva, unsigned long size,
 
 	trace_sp_k2u_begin(kc);
 
-	if (sp_flags & ~SP_DVPP) {
+	if (sp_flags & ~SP_FLAG_MASK) {
 		pr_err_ratelimited("k2u sp_flags %lx error\n", sp_flags);
 		return -EINVAL;
 	}
+	sp_flags &= ~SP_HUGEPAGE;
 
 	if (!current->mm) {
 		pr_err_ratelimited("k2u: kthread is not allowed\n");
-- 
2.17.1

